src/rgw/rgw_lua_utils.cc

```diff
@@ -25,6 +26,74 @@ int RGWDebugLog(lua_State* L)
   return 0;
 }
 
+int SetAttribute(lua_State* L)  {
+  auto s = reinterpret_cast<req_state*>(lua_touserdata(L, lua_upvalueindex(1)));
+
+  if (!s->trace || !s->trace->IsRecording()) {
+    return 0;
+  }
+
+  auto key = luaL_checkstring(L, 1);
+  int value_type = lua_type(L, 2);
+
+  switch (value_type) {
```

Comment:
i think that you can change that:
* check if the value is a number, and convert accordingly
* if not, just call `to_string()` - i think that anything can be converted to a string, so we won't need the error case